src/js/jsx/shared/Datalist.jsx

```diff
@@ -314,7 +316,8 @@ define(function (require, exports, module) {
                 case "Tab":
                     return;
                 case "Enter":
-                case "Return":
```

Comment:
My guess is that you'll want to remove this code once you get the `defaultOptions` working because this code path is only active when the `Select` component is _not_ being shown.